Rename ad578x driver

SaikiranGudla opened the pull request:
Rename ad578x driver, files and other occurances
to ad579x

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Hardware:
* OS:

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

SaikiranGudla commented:
Is this PR good to be merged, @tfcollins ?

tfcollins commented:
What is this being renamed? Does it not support ad5780 devices?

SaikiranGudla commented:
> Is this PR good to be merged, @tfcollins ?



> What is this being renamed? Does it not support ad5780 devices?

It is being renamed to ad579x as it's the flagship part and also to align with no-os, FW and linux drivers. It still supports ad578x, 6x, 9x devices.

SaikiranGudla commented:
Is this PR good to be merged, @tfcollins ?

tfcollins commented:
@SaikiranGudla please rebase onto main

tfcollins reviewed (changes requested):
Does this affect https://github.com/analogdevicesinc/pyadi-iio/blob/main/supported_parts.md ?
Please update this file as well https://github.com/analogdevicesinc/pyadi-iio/blob/main/test/emu/hardware_map.yml#L450

SaikiranGudla commented:
> @SaikiranGudla please rebase onto main

Rebased onto main

SaikiranGudla commented:
> Does this affect https://github.com/analogdevicesinc/pyadi-iio/blob/main/supported_parts.md ? Please update this file as well https://github.com/analogdevicesinc/pyadi-iio/blob/main/test/emu/hardware_map.yml#L450

No, the supported parts will still be the same.
Updated hardware_map.yml, renamed xml file.

tfcollins commented:
If you can just update the supported_parts.md file we are good

SaikiranGudla commented:
> If you can just update the supported_parts.md file we are good

The supported_parts.md file already contains all the supported parts (AD5760/80/81/90/91)
https://github.com/analogdevicesinc/pyadi-iio/blob/main/supported_parts.md?plain=1#L63

tfcollins commented:
@SaikiranGudla as you can see CI is failing. Please add ad579x here https://github.com/analogdevicesinc/pyadi-iio/blob/main/tasks.py#L164
